user/tests/tests_view.py

```diff
@@ -0,0 +1,114 @@
+
+    def test_user_detail_view(self):
+        user_instance = self.create_user()
+        url = reverse('user-detail', args=[user_instance.id])
+        resp = self.client.get(url)
+        self.assertEqual(resp.status_code, 200)
+        json_resp = json.loads(resp.content)
+        self.assertDictContainsSubset(model_to_dict(user_instance, MODEL_FIELDS), json_resp, 'it should give detail for the user asked with id')
+
+    def test_user_login(self):
+        self.create_user()
+        login_resp = self.client.post(
+            '/api/token', {'username': test_user['username'], 'password': test_user['password']})
+        self.assertEqual(login_resp.status_code, 200, 'it should login')
+
+    def test_loggedin_user_detail_view_without_token(self):
+        resp = self.client.get('/api/user/')
+        self.assertEqual(resp.status_code, 401, 'it should not give any detail of user')
+
+    def test_loggedin_user_detail_view(self):
+        user_instance = self.create_user()
+        login_resp = self.client.post(
+            '/api/token', {'username': test_user['username'], 'password': test_user['password']})
+        login_json_resp = json.loads(login_resp.content)
+        token = login_json_resp['access']
+        resp = self.client.get(
+            '/api/user/', HTTP_AUTHORIZATION='Bearer ' + token)
+        self.assertEqual(resp.status_code, 200)
+        json_resp = json.loads(resp.content)
+        self.assertDictContainsSubset(model_to_dict(user_instance, MODEL_FIELDS), json_resp, 'it should give detail for logged in user')
+
+    def test_user_register_view(self):
+        user_instance = self.create_user()
+        resp = self.client.post('/api/register/',
+                                {'username': 'admin', 'password': '12345678', 'email': 'admin@test.com'})
+        json_resp = json.loads(resp.content)
+        self.assertEqual(resp.status_code, 201)
+        self.assertNotEqual(str(user_instance.id), json_resp['id'], 'it should register new user')
+
+    def test_user_update_view_without_token(self):
+        user_instance = self.create_user()
+        data = json.dumps({'first_name': 'update first name'})
+        resp = self.client.patch(
+            '/api/user/', data, content_type='application/json')
+        # 401 status code will be sent by the backend for unauthorized requests
+        self.assertEqual(resp.status_code, 401, 'it should not allow user to update its content without token')
+
+    def test_user_update_view_with_token(self):
+        user_instance = self.create_user()
+        login_resp = self.client.post(
+            '/api/token', {'username': test_user['username'], 'password': test_user['password']})
+        self.assertEqual(login_resp.status_code, 200)
+        login_json_resp = json.loads(login_resp.content)
+        token = login_json_resp['access']
+        data = json.dumps({'first_name': 'update first name'})
+        user_resp = self.client.patch('/api/user/', data,
```

Comment:
this should be a utility function (

Reply:
you mean self.client.patch should be a utility function?